lib/Runtime/Debug/TTRuntmeInfoTracker.cpp

```diff
@@ -491,22 +489,71 @@ namespace TTD
         return this->m_ttdFunctionBodyParentMap.LookupWithKey(body, nullptr);
     }
 
-    Js::FunctionBody* ScriptContextTTD::FindFunctionBodyByFileName(const char16* filename) const
+    uint64 ScriptContextTTD::FindTopLevelCtrForBody(Js::FunctionBody* body) const
+    {
+        Js::FunctionBody* rootBody = body;
+        while(this->ResolveParentBody(rootBody) != nullptr)
+        {
+            rootBody = this->ResolveParentBody(rootBody);
+        }
+
+        TTD_PTR_ID trgtid = TTD_CONVERT_FUNCTIONBODY_TO_PTR_ID(rootBody);
+
+        for(int32 i = 0; i < this->m_ttdTopLevelScriptLoad.Count(); ++i)
+        {
+            if(this->m_ttdTopLevelScriptLoad.Item(i).ContextSpecificBodyPtrId == trgtid)
+            {
+                return this->m_ttdTopLevelScriptLoad.Item(i).TopLevelBodyCtr;
+            }
+        }
+
+        for(int32 i = 0; i < this->m_ttdTopLevelNewFunction.Count(); ++i)
+        {
+            if(this->m_ttdTopLevelNewFunction.Item(i).ContextSpecificBodyPtrId == trgtid)
```

Comment:
The function body should have flags indicating whether it's a regular function or was created as part of eval/new function. Should we assert that the body has the appropriate flags set?